pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -293,17 +302,23 @@ func (c *Reconciler) reconcile(ctx context.Context, tr *v1alpha1.TaskRun) error
 		pod, err = c.createPod(tr, rtr.TaskSpec, rtr.TaskName)
 		if err != nil {
 			// This Run has failed, so we need to mark it as failed and stop reconciling it
-			var msg string
-			if tr.Spec.TaskRef != nil {
-				msg = fmt.Sprintf("References a Task %s that doesn't exist: ", fmt.Sprintf("%s/%s", tr.Namespace, tr.Spec.TaskRef.Name))
+			var reason, msg string
+			if isExceededResourceQuotaError(err) {
+				reason = reasonExceededResourceQuota
+				msg = getExceededResourcesMessage(tr)
 			} else {
-				msg = fmt.Sprintf("References a TaskSpec with missing information: ")
+				reason = reasonCouldntGetTask
+				if tr.Spec.TaskRef != nil {
+					msg = fmt.Sprintf("References a Task %s/%s that doesn't exist", tr.Namespace, tr.Spec.TaskRef.Name)
+				} else {
+					msg = fmt.Sprintf("References a TaskSpec with missing information")
```

Comment:
this is outside the scope of your change, but i think that in spite of what the previous logic indicated (which I may have added... 😇 ) I think there are a variety of things that could have gone wrong here (if i remember right, even templating problems) - so one option would be to make this super generic, e.g. something like "invalid task"?